File: includes/adj_list.py
from includes.team import Team
from includes.game import Game

class AdjList():

    def __init__(self):
        self.head, self.tail = None, None
        self._size = 0

    def getSize(self): return self._size

    def insertHead(self, name):
        self.tail = self.head
        self.head = Team(name)
        if self.tail == None:
            self.tail = self.head
        self._size += 1

    def insert(self, team2, scoreDiff):
        curr = self.tail
        curr.next = Game(team2, scoreDiff)
        self.tail = curr.next
        self._size += 1

File: cpp/src/adj_list.cpp
#include "adj_list.h"

AdjList::AdjList()
{
    head_ = NULL;
    tail_ = NULL;
    size_ = 0;
}

int AdjList::getSize()
{
    return size_;
}

void AdjList::insertHead(string name)
{
    tail_ = head_;
    head_ = new Team(name);
    if (tail_ == NULL)
    {
        tail_ = head_;
    }
    size_ += 1;
}

void AdjList::insert(string team2, int scoreDiff)
{
    curr_ = tail_;
    curr_->next() = new Game(team2, scoreDiff);
    tail_ = curr_->next();
    size_ += 1;
}

Team* AdjList::getHead()
{
    return head_;
}

Game* AdjList::getTail()
{
    return tail_;
}
